coreppl/dist.mc

```diff
@@ -178,6 +249,64 @@ lang BernDist = Dist + PrettyPrint + Eq + Sym + BoolTypeAst + FloatTypeAst
 
 end
 
+
+
+
+lang PoissonDist = Dist + PrettyPrint + Eq + Sym + BoolTypeAst + FloatTypeAst
+
+  syn Dist =
+  | DPoisson { lambda: Expr }
+
+  sem smapDist_Expr_Expr (f: Expr -> a) =
+  | DPoisson t -> DPoisson { t with lambda = f t.lambda }
+
+  sem sfoldDist_Expr_Expr (f: a -> b -> a) (acc: a) =
+  | DPoisson t -> f acc t.lambda
+
+  -- Pretty printing
+  sem pprintDist (indent: Int) (env: PprintEnv) =
+  | DPoisson t ->
+    let i = pprintIncr indent in
+    match printParen i env t.lambda with (env,lambda) then
+      (env, join ["Poisson", pprintNewline i, lambda])
+    else never
+
+  -- Equality
+  sem eqExprHDist (env : EqEnv) (free : EqEnv) (lhs : Expr) =
+  | DPoisson r ->
+    match lhs with DPoisson l then eqExprH env free l.lambda r.lambda else None ()
+
+  -- Symbolize
+  sem symbolizeDist (env: SymEnv) =
+  | DPoisson t -> DPoisson { t with lambda = symbolizeExpr env t.lambda }
+
+  -- Type Annotate
+  sem tyDist (env: TypeEnv) (info: Info) =
+  | DPoisson t ->
+    match ty t.lambda with TyFloat _ then TyBool { info = NoInfo () }
```

Comment:
Shouldn't the type for Poisson be TyInt instead of TyBool?

Reply:
Good catch, thanks! Fixed!